common4j/src/main/com/microsoft/identity/common/java/util/CommandLineParameter.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.util;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Static class for accessing command line parameters.
+ * */
+public class CommandLineParameter {
```

Comment:
Why do we need this? 

If the class where these (slice, and dc) are set is already moved to common4j, then we shouldn't a platform-dependent way to set these...these are just gradle properties...we should be able to set them directly from the BuildConfig in common4j. Any consumers of common4j (common Android, Linux Broker) shouldn't care about these and they don't need to provide a way to set these.

Reply:
AFAIK, BuildConfig is an Android thing. There's a plugin for java (https://github.com/mfuerstenau/gradle-buildconfig-plugin), but it hasn't been updated since early 2017.

Reply:
We could figure out an xplat way to set this value - if you have any suggestion (and it's small), I can do it now. Otherwise I could create a PBI for this.

Reply:
We are using this one: https://github.com/gmazzo/gradle-buildconfig-plugin (updated 20 days ago)

READ THIS: https://github.com/AzureAD/android-complete/blob/master/docs/ProjectBuild/build_config.md

I already did it for broker4j...I might not have done it yet for common4j...but I think we should do that, and then we don't need a class called CommandLineParameter

Reply:
ok. will do. Thanks!

Reply:
Done.